app/controllers/customers_controller.rb

```diff
@@ -0,0 +1,12 @@
+class CustomersController < ApplicationController
+
+    def show
+        @customer = Customer.find(params[:id])
+        if params[:item_id].present?
```

Comment:
This implementation will work, but it leaves this controller method with a lot of responsibility.

Another, more RESTful and SRP-friendly solution would be putting the content inside this if statement (creating the joins record) in a different controller action. 

What you are doing in this if statement is **creating** the record in the joins table to associate an item with a customer. So, it would make sense to put that functionality in a `create` action. But, we don't want to put the `create` action in this controller, because that should really be reserved for when we create customer objects. 

So, a conventional way to do this following RESTful principles is to create a joins controller: the CustomerItemsController. In this controller, you could add a `create` action that includes `CustomerItem.create!(customer_id: @customer.id, item_id: @item.id)` and then redirects back to this customer show page. 


There are times when we route a form back to the same show action, but that's usually when we are just manipulating data that we are already rendering on the page (i.e. we are reading data based on a search input, or we are ordering the same data alphabetically). In this case, we are doing a DB operation with this joins record, so it's best to create a separate controller action. 

Let me know if you have questions on this! I know this comment is super long. :)